Source/SwiftLintFramework/Models/Configuration.swift

```diff
@@ -78,17 +79,44 @@ public struct Configuration: Equatable {
         }
         self.disabledRules = validDisabledRules
 
-        self.rules = rules.filter { rule in
-            let id = rule.dynamicType.description.identifier
-            if validDisabledRules.contains(id) { return false }
-            return enabledRules.contains(id) || !(rule is OptInRule)
+        // white_list rules take precendence over all else.
+        if !whitelistRules.isEmpty {
+
+            if !disabledRules.isEmpty || !optInRules.isEmpty {
+                queuedPrintError("'disabled_rules' or 'opt_in_rules' cannot be used in " +
+                    "combination with 'whitelist_rules'")
+                return nil
+            }
+
+            self.rules = rules.filter { rule in
+                let id = rule.dynamicType.description.identifier
+                return whitelistRules.contains(id)
+            }
+        } else {
+            self.rules = rules.filter { rule in
+                let id = rule.dynamicType.description.identifier
+                if validDisabledRules.contains(id) { return false }
+                return optInRules.contains(id) || !(rule is OptInRule)
+            }
         }
     }
 
     public init?(dict: [String: AnyObject]) {
+
+        // Deprecation warning for "enabled_rules"
+        if dict["enabled_rules"] != nil {
+            queuedPrint("'enabled_rules' has been renamed to 'opt_in_rules' " +
+                "and will be completely removed in a future release.")
```

Comment:
This is a tiny nitpick, but might be better to say something like "the behavior of `enabled_rules` will be changing in a future release. Use `opt_in_rules` instead.`" I only say this because it might be confusing for someone to think `enabled_rules` is going away completely and then is told to use it for something else. Unless that is we decided to use a different label for the whitelist altogether?